components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarFeature.kt

```diff
@@ -7,26 +7,31 @@ package mozilla.components.feature.toolbar
 import mozilla.components.concept.toolbar.Toolbar
 import mozilla.components.browser.session.SessionManager
 
+typealias LoadUrlUseCase = (String) -> Unit
```

Comment:
This is interesting. :)

I wonder if this is still readable/understandable when the functions get a bit more complicated, e.g.:
```Kotlin
(String, String, Boolean, Int) -> Unit.
```

But then again Kotlin seems to accept named parameteres in the typealias too:
```Kotlin
(firstname: String, lastname: String, hasJob: Boolean, age: Int) -> Unit
```

^-- But they are not enforced and the IDE only tells you that it needs to be String,String,Boolean,Int :(

In addition to that if you pass a function (e.g. foo(::bar)) and you do not use all parameters in bar then the IDE complaints about unused parameters - because it does not know that *bar* implements a specific typealias signature.

When changing the typealias then the errors the IDE shows are not really helpful. With interfaces it knows exactly what is wrong and offers to add missing things. :)